code/game/objects/items/marine_gear.dm

```diff
@@ -246,6 +246,10 @@
 	time_to_unequip = 1 SECONDS
 	flags_inventory = NOQUICKEQUIP
 
+/obj/item/belt_harness/on_vend(faction)
+	if (!usr.equip_to_appropriate_slot(src))
+		..()
```

Comment:
do not use usr
preserve parent return, use return ..() pls
applies for all other instances of this

Reply:
Removed all uses of usr in on_vend, and modified all on_vend overrides to preserve parent return